app/subscriber/src/features/manage-folder/ManageFolder.tsx

```diff
@@ -0,0 +1,171 @@
+import { SearchWithLogout } from 'components/search-with-logout';
+import { DetermineToneIcon } from 'features/home';
+import { determinePreview } from 'features/utils';
+import parse from 'html-react-parser';
+import React from 'react';
+import { DragDropContext, Draggable, Droppable } from 'react-beautiful-dnd';
+import { FaArrowLeft, FaFolderMinus, FaGripLines } from 'react-icons/fa';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useContent } from 'store/hooks';
+import { useFolders } from 'store/hooks/subscriber/useFolders';
+import { Checkbox, Col, IContentModel, IFolderModel, Row } from 'tno-core';
+
+import * as styled from './styled';
+
+/**
+ * Component that renders the manage folder page, allowing user to view/manage content inside a folder
+ * @returns ManageFolder component
+ */
+export const ManageFolder: React.FC = () => {
+  const { id } = useParams();
+  const [, { getFolder, updateFolder }] = useFolders();
+  const navigate = useNavigate();
+  const [, { findContent }] = useContent();
+  const [folder, setFolder] = React.useState<IFolderModel>();
+  const [items, setItems] = React.useState<any>([]);
+  const [selected, setSelected] = React.useState<IContentModel[]>([]);
+
+  /** TODO: Folder content only contains contentId and sortOrder so we have to make an additional call based off of the contentIds to get the headline/summary etc..
+   * assuming we want this to differ eventually.
+   */
+  React.useEffect(() => {
+    getFolder(Number(id)).then((folder) => {
+      // have to use temp not folder state as folder state will not update right away
+      const temp = folder;
+      setFolder(folder);
+      findContent({
+        contentIds: folder.content.map((c) => c.contentId),
+      })
+        .then((data) => {
+          const tempSort = data.items.map((item) => ({
+            ...item,
+            sortOrder: temp.content.find((c) => c.contentId === item.id)?.sortOrder ?? 0,
+          }));
+          tempSort.sort((a, b) => a.sortOrder - b.sortOrder);
+          setItems(tempSort);
+        })
+        .catch();
+    });
+    // Only on initialize, or when sort order changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /** function that runs after a user drops an item in the list */
+  const handleDrop = React.useCallback(
+    async (droppedItem: any) => {
```

Comment:
You need to include `try+catch` just like you need a `.catch` for promises.  If you open your developer tools and execute this code it should give you a warning (normally in red) saying an unhandled promise.